.github/workflows/docker-publish.yml

```diff
@@ -58,85 +61,52 @@ jobs:
           --build-arg PIPELINE_PACKAGE=${{ env.PIPELINE_PACKAGE }} \
           --provenance=false \
           --progress plain \
-          --cache-from $DOCKER_BUILD_REPOSITORY:amd \
-          -t $DOCKER_BUILD_REPOSITORY:amd-$SHORT_SHA .
+          --cache-from $DOCKER_BUILD_REPOSITORY:$ARCH \
+          -t $DOCKER_BUILD_REPOSITORY:$ARCH-$SHORT_SHA .
     - name: Set virtualenv cache
       uses: actions/cache@v3
       id: virtualenv-cache
       with:
         path: |
           .venv
         key: ci-venv-${{ env.PIPELINE_FAMILY }}-${{ hashFiles('requirements/test.txt') }}
-    - name: Test AMD image
-      run: |
-        source .venv/bin/activate
-        DOCKER_IMAGE="$DOCKER_BUILD_REPOSITORY:amd-$SHORT_SHA" make docker-test 
-    - name: Push AMD image
-      run: |
-        # write to the build repository to cache for the publish-images job
-        docker push $DOCKER_BUILD_REPOSITORY:amd-$SHORT_SHA
-  build-arm:
-    runs-on: ubuntu-latest
-    needs: [setup, set-short-sha]
-    env:
-      SHORT_SHA: ${{ needs.set-short-sha.outputs.short_sha }}
-    steps:
-    - name: Set up Docker Buildx
-      uses: docker/setup-buildx-action@v2
-    - name: Checkout code
-      uses: actions/checkout@v3
-    - name: Login to Quay.io
-      uses: docker/login-action@v2
-      with:
-        registry: quay.io
-        username: ${{ secrets.QUAY_IO_ROBOT_USERNAME }}
-        password: ${{ secrets.QUAY_IO_ROBOT_TOKEN }}
     - name: Set up QEMU
       uses: docker/setup-qemu-action@v2
-    - name: Build ARM image
-      run: |
-        DOCKER_BUILDKIT=1 docker buildx build --platform=linux/arm64 --load \
-          --build-arg PIP_VERSION=$PIP_VERSION \
-          --build-arg BUILDKIT_INLINE_CACHE=1 \
-          --build-arg PIPELINE_PACKAGE=${{ env.PIPELINE_PACKAGE }} \
-          --provenance=false \
-          --progress plain \
-          --cache-from $DOCKER_BUILD_REPOSITORY:arm \
-          -t $DOCKER_BUILD_REPOSITORY:arm-$SHORT_SHA .
-    - name: Set virtualenv cache
-      uses: actions/cache@v3
-      id: virtualenv-cache
-      with:
-        path: |
-          .venv
-        key: ci-venv-${{ env.PIPELINE_FAMILY }}-${{ hashFiles('requirements/test.txt') }}
-    - name: Test ARM image
+    - name: Test image
       run: |
+        ARCH=$(cut -d "/" -f2 <<< ${{ matrix.docker-platform }})
         source .venv/bin/activate
-        # skip the inference tests because they will hang on emulated architecture
-        SKIP_INFERENCE_TESTS=true DOCKER_IMAGE="$DOCKER_BUILD_REPOSITORY:arm-$SHORT_SHA" make docker-test 
-    - name: Push ARM image
+        if [ "${{ matrix.docker-platform }}" == "linux/arm64" ]; then
+          DOCKER_PLATFORM="${{ matrix.docker-platform }}" DOCKER_IMAGE="$DOCKER_BUILD_REPOSITORY:$ARCH-$SHORT_SHA" \
+          SKIP_INFERENCE_TESTS=true make docker-test
+        else
+          DOCKER_PLATFORM="${{ matrix.docker-platform }}" DOCKER_IMAGE="$DOCKER_BUILD_REPOSITORY:$ARCH-$SHORT_SHA" \
+          make docker-test
```

Comment:
nit: you can put `DOCKER_PLATFORM="${{ matrix.docker-platform }}" DOCKER_IMAGE="$DOCKER_BUILD_REPOSITORY:$ARCH-$SHORT_SHA"` before the conditional since this is just the same in both cases, right?

Reply:
It is, I am running into an issues where it's not liking the prefix variables being set as part of a command array.  It works if I eval, but I try to stay away from that from a security standpoint.  Going to leave this for now, unless you have any suggestions here @ryannikolaidis

Reply:
I might be confused. I'm not sure how what you have is functionally different from:

```
        DOCKER_PLATFORM="${{ matrix.docker-platform }}"         DOCKER_IMAGE="$DOCKER_BUILD_REPOSITORY:$ARCH-$SHORT_SHA" \
        if [ "${{ matrix.docker-platform }}" == "linux/arm64" ]; then
          SKIP_INFERENCE_TESTS=true make docker-test
        else
          make docker-test
```

Reply:
ohh nvm, I see what you're saying, you're calling both in a single call. okay, let me see if I have any thoughts

Reply:
oh huh! does exporting the env vars work? yea, definitely agree on no-eval for security reasons. also it's a super nit, so no need to block on this, just curious

Reply:
ohh yea, just wondering how to minimize duplication without sacrificing readability. but yea, def not worth sweating over!